contributing/tests.md

```diff
@@ -7,268 +7,52 @@ describes them and gives pointers on how to run/debug them.
-
-1. Build all the modules, like normal, and write their JS files and source maps
-   to disk.
-1. Use `istanbul` (version 1.0 and above) that supports source maps to
-   instrument the newly build modules from step 1 and write it out to disk.
-1. `istanbul` works by source instrumentation and maintaining a global map of
-   files to covered locations called `__coverage__`.
-1. At the end of our extension deactivation, we grab the value of the
-   `__coverage__` variable and write it out into coverage/coverage.json
-1. The `coverage/coverage.json` file still uses the JS line numbers. We then use
-   `remap-istanbul` to remap the line numbers to their TypeScript counterparts.
-1. Then we replace the original `coverage/coverage.json` with this new version
-   and generate the corresponding coverage.lcov.
-
-Because we are changing the built modules and writing to file, the steps above
-have to be explicitly requested using the command `npm run coverage:system-tests`.
-
-Tip: Depending on what you are trying to do, it is likely that you need to
-rebuild the modules after the coverage tests to get to a pristine state.
+Unit tests priorities are as follows:
+
+1. Unit tests should focus on only the _unit_ of code under test.
+   - 1 method/function
+1. All dependancies should be stubbed/mocked.
+1. Unit tests should be easy to write and execute
+   - If your code is hard to test consider how it could be refactored to make it easier to test.
+1. Use only jest and do not import mochi/sinon/chai.
+1. Code coverage is one measure of how well we are unit testing our code. We should strive to code all code paths for any touched code in the repository. Code coverage can be generate by including the `--coverage` flag to jest when executing.
+
+#### How to Write Jest Unit Tests
+
+- Test files use the {fileUnderTestName}.test.ts format and should go under the same directory structure as the source in the test/jest folder.
+  - Note this is to separate from the existing mocha unit tests and enable us to track our progress for moving off mocha to jest.
+- VSCode extensions are our friend
+  - https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner
+- Tests can be executed from the IDE, command line, or via npm script.
+  - IDE assuming you are using the above extension use the Run & Debug code lense found in test files.
+  - Command line: `npx jest` will execute all unit tests in a package by default. If the integration tests are configured for jest in the package you may execute those using `npx jest -c jest.integration.config`. There are many options for configuring the test run see the [docs](https://jestjs.io/docs/cli) or `npx jest -h` for further guidance.
+  - npm scripts:
+    - `npm run test:unit`
+    - `npm run test:integration`
+    - `npm run test`
+
+#### Jest Information
+
+- ts-jest enables us to compile on the fly so no need for compilation while writing unit tests.
```

Comment:
Does one have to run an npm command to get the on-the-fly compiling working?

Reply:
nope...it's baked into the jest configuration so nothing needed to run jest with ts compilation

Reply:
Looks great! Just caught one small typo, fix -- "Code coverage can be generate" to "generated".

Reply:
thanks!